core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,240 @@
+        cancelInitiated = true;
+    }
+
+    /**
+     * Enable client to subscribe and receive all the responses.
+     * @return Return poll response as Flux
+     */
+    public Flux<PollResponse<T>> getObserver() {
+         createFlux();
+         return fluxHandle;
+    }
+
+    /**
+     * Calls poll operation once in sync.
+     * @return  Mono of poll response
+     */
+    public Mono<PollResponse<T>> poll() {
+        updatePollOperationSync();
+        return Mono.just(pollResponse);
+    }
+
+    /**
+     * This will block till poll operation is complete
+     * @return returns poll response
+     */
+    public PollResponse<T> block() {
+        return getObserver().blockLast();
+    }
+
+    /*
+     * Calls poll operation function and update pollResponse.
+     */
+    private  void updatePollOperationSync() {
+        pollResponse = pollOperation.apply(pollResponse);
+    }
+
+    /*
+     * Get whether or not this PollStrategy's long running operation is done.
+     * @return Whether or not this PollStrategy's long running operation is done.
+     */
+    private Mono<PollResponse<T>> sendPollRequestWithDelay() {
+        return Mono.defer(() -> delayAsync().then(Mono.defer(() -> {
+            if (!isTerminalState()) {
+                updatePollOperationSync();
+            } else if (!isTerminalState()) {
+                return Mono.empty();
+            }
+            return Mono.just(pollResponse);
+        })));
+    }
+
+    /*
+     * Helper function
+     */
+    private boolean isTerminalState() {
+        return pollResponse != null && pollResponse.isDone();
+
+    }
+
+    /*
+     * If this PollerOptions has a pollIntervalInMillis value, return an Mono that is delayed by the
+     * number of seconds that are in the pollIntervalInMillis value. If this PollerOptions doesn't have
+     * a pollIntervalInMillis value, then return an Single with no delay.
+     *
+     * @return A Mono with delay if this PollerOptions has a pollIntervalInMillis value.
+     */
+    private Mono<Void> delayAsync() {
+        Mono<Void> result = Mono.empty();
+        if (this.pollerOptions.pollInterval().toNanos() > 0) {
+            result = result.delaySubscription(this.pollerOptions.pollInterval());
+        }
+        return result;
+    }
+
+    /**
+     *
+     * @param  autoPollingEnabled  true : Ensures the polling is happening in background.
+     *                             false : Ensures that polling is not happening in background.
+     */
+    public void setAutoPollingEnabled(boolean autoPollingEnabled) {
+    	
+    	// setting same auto polling status would not require any action.
+    	
+    	if (this.autoPollingEnabled == autoPollingEnabled ) {
+    		return;
+    	}
+
+    	this.autoPollingEnabled = autoPollingEnabled;
+        if (this.autoPollingEnabled) {
+            if (valid(fluxDisposable)) {
+                fluxDisposable = fluxHandle.subscribe(pr ->  pollResponse = pr );
```

Comment:
I understand what you are trying to do here. In the case of auto-polling, you want to kick off the auto-polling `Flux`. This works but is inconsistent with other Publishers in our libraries because this will become a "hot" publisher. 

We want to do it in the way that only when a user subscribes to our `fluxHandle` then it automatically starts polling. I imagine something like this:

```java
startCopyStorageBlob(blob1url, blob2url)
.map(res -> createPollingResponse(res))
.map(pollRes -> new Poller(pollRes, pr -> createPollingResponse(checkCopyStorageBlobStatus(pr.getValue()))))
.flatMap(poller -> poller.autoPoll())
.filter(pollRes.isDone())
.next()
.subscribe(pollRes -> System.out.println("polling done " + pollRes.value));

Reply:
The way I understand that Arch board has decided to "auto-polling by default "  . @JonathanGiles  please confirm.

Reply:
Polling will happen automatically when the poller is created, regardless of the user subscribing or not